lib/shimmer.js

```diff
@@ -508,62 +535,70 @@ function applyDebugState(shim, nodule) {
  * @param resolvedName
  */
 function instrumentPostLoad(agent, nodule, moduleName, resolvedName) {
-  const instrumentation = shimmer.registeredInstrumentations[moduleName]
-  if (
-    properties.hasOwn(nodule, symbols.instrumented) ||
-    properties.hasOwn(nodule, symbols.instrumentedErrored)
-  ) {
-    logger.trace(
-      'Already instrumented or failed to instrument %s, skipping redundant instrumentation',
-      moduleName
+  const instrumentations = shimmer.registeredInstrumentations[moduleName]
+  instrumentations.forEach((instrumentation) => {
+    if (
+      properties.hasOwn(instrumentation, symbols.instrumented) ||
+      properties.hasOwn(instrumentation, symbols.instrumentedErrored)
+    ) {
+      logger.trace(
+        'Already instrumented or failed to instrument %s, skipping redundant instrumentation',
+        moduleName
+      )
+      return
+    }
+
+    const shim = shims.createShimFromType(
+      instrumentation.type,
+      agent,
+      moduleName,
+      resolvedName,
+      instrumentation.shimName
     )
-    return nodule
-  }
 
-  const shim = shims.createShimFromType(instrumentation.type, agent, moduleName, resolvedName)
-
-  applyDebugState(shim, nodule)
-  trackInstrumentationUsage(
-    agent,
-    shim,
-    instrumentation.specifier || moduleName,
-    NAMES.FEATURES.INSTRUMENTATION.ON_REQUIRE
-  )
-
-  // Tracking instrumentation is only used to add the supportability metrics
-  // that occur directly above this.  No reason to attempt to load instrumentation
-  // as it does not exist.
-  if (instrumentation.type === MODULE_TYPE.TRACKING) {
-    return nodule
-  }
+    applyDebugState(shim, nodule)
+    trackInstrumentationUsage(
+      agent,
+      shim,
+      instrumentation.specifier || moduleName,
+      NAMES.FEATURES.INSTRUMENTATION.ON_REQUIRE
+    )
 
-  try {
-    if (instrumentation.onRequire(shim, nodule, moduleName) !== false) {
-      nodule = shim.getExport(nodule)
-      nodule[symbols.instrumented] = true
+    // Tracking instrumentation is only used to add the supportability metrics
+    // that occur directly above this.  No reason to attempt to load instrumentation
+    // as it does not exist.
+    if (instrumentation.type === MODULE_TYPE.TRACKING) {
+      return
```

Comment:
hmm I should set the instrumentation[symbols.instrumented] property to true.  Even if it didn't instrument anything this is an indicator in instrumentPostLoad which tells api.instrumentedLoadedModule if it succeeded meaning every instrumentation hook was instrumented